lightning/src/onion_message/messenger.rs

```diff
@@ -170,8 +170,8 @@ impl<Signer: Sign, K: Deref, L: Deref> OnionMessenger<Signer, K, L>
 			packet_payloads, packet_keys, prng_seed).map_err(|()| SendError::TooBigPacket)?;
 
 		let mut pending_per_peer_msgs = self.pending_messages.lock().unwrap();
-		let pending_msgs = pending_per_peer_msgs.entry(introduction_node_id).or_insert(Vec::new());
-		pending_msgs.push(
+		let pending_msgs = pending_per_peer_msgs.entry(introduction_node_id).or_insert(VecDeque::new());
```

Comment:
Use `or_insert_with(VecDeque::new)`. No need to create a new `VecDeque` if not needed especially given `VecDeque::new` will allocate space whereas `Vec::new` won't.

Reply:
That’s surprising behavior but seems to be the case. In a previous version I had an additional peer_set Messenger struct field — this would allow us to only allocate on OM send, rather than on initial peer connection. Thoughts on whether this would be preferable?

Reply:
Ah I thought this was a comment on #1650, so my response applies there… yes, I’ll make this change here, thanks